### PR TITLE
Feat/configurable paths

### PR DESCRIPTION
**Pull request by suchmememanyskill**
the tl;dr is that this removes the PLUGIN_PATH env var (gets read for backwards compat still) and adds PRIVILEGED_PATH, UNPRIVILEGED_PATH and UNPRIVILEGED_USER
by default, the following fallbacks are used, in the following order:

PRIVILEGED_PATH:
1. UNPRIVILEGED_PATH

UNPRIVILEGED_PATH
1. parent dir of PLUGIN_PATH
2. parent dir of parent dir of binary location
3. '/home/deck/homebrew'

UNPRIVILEGED_USER
1. check .startswith of UNPRIVILEGED_PATH against available pw_dirs
2. 'deck' 

For windows, fallbacks are different
PRIVILEGED_PATH instantly redirects to UNPRIVILEGED_PATH

UNPRIVILEGED_PATH:
1. PRIVILEGED_PATH
2. %userprofile%/homebrew

UNPRIVILEGED_USER
1. currently logged in user 

image

### CODE REVIEW

**Comment by TrainDoctor** [comment]
@SteamDeckHomebrew/decky-testers please give this a try and report any issues.

**Review by TrainDoctor — APPROVED** [review]
Works great during testing. Merging and sending to pre-release.
